pkg/scheduler/apis/config/types.go

```diff
@@ -55,6 +55,9 @@ const (
 type KubeSchedulerConfiguration struct {
 	metav1.TypeMeta
 
+	// CPU threads
+	CPUThreads int
```

Comment:
There isn't a one-to-one correspondence between real CPU threads and the parallelism we're talking about here. Goroutines are multiplexed over OS threads/processes that is configurable with `GOMAXPROCS` (which defaults to the number of cores available).

We should probably define this `Parallelism int` to not mislead users. As an example to show the difference between the different things involved here, the user can set `GOMAXPROCS=4` and `Parallelism = 16` in a machine with 8 physical CPUs -- the actual number of CPU threads in this example is 4.

Reply:
I agree, but a universal solution may require more simulations. For now, I have updated the PR, there were some hunks missing. I was testing some changes and this got converted from draft. Right now the pressing challenge is to get the currently loaded configuration of the scheduler on parallelism as I see that the config is propagated via arguments.

Reply:
No, I meant parallelism within the scheduler, not parallelism elsewhere i.e., change the parameter's name from `CPUThreads` to `Parallelism` so that we don't mislead users of `KubeSchedulerConfiguration`.

Reply:
+1 to `Parallelism` or `Routines`, but I prefer the first one. Note that this name is part of an API, and as such, it can be changed easily. Also a good comment is worth. We don't want people to think this is about several Pods being scheduled at the same time.

Maybe even `AlgorithmsParallelism` is more appropriate, but I'm not convinced.

Reply:
Actually, upon further thought, I think `Concurrency` (or `AlgorithmsConcurrency`) might be better since goroutines are only concurrent, not necessarily parallel.

Reply:
That's a bit nit-picky on implementation details.

In their most common interpretation, Concurrency is about dealing with lots of things at once (like multiple unrelated requests, or multiple Pods in k8s world), whereas parallelism is about doing lots of things at once, like subtasks (so more suited to talk about algorithms).

Reply:
Let's make it `Parallelism`